File: other/Solution1004.java
class Solution1004 {
    public int longestOnes(int[] A, int K) {
    	int left = 0;
    	int right = 0;
    	int max = 0;
    	int count = 0;
    	while(right < A.length){
    		if(A[right] == 1){
    			count += 1;
    		}
    		int length = right - left + 1;
    		if(count + K < length){
    			left += 1;
    			if(A[left - 1] == 1){
    				count -= 1;
    			}
    		}
    		length = right - left + 1;
    		max = Math.max(length, max);
    		right += 1;
    	}
    	return max;
    }
}

File: C++/Solution1004.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution1004 {
public:
    int longestOnes(vector<int>& A, int K) {
        int size = A.size();
        int left = 0, right = 0;
        int count = A[0];
        int len = 0;
        while(right < size){
            if(count + K >= right - left + 1){
                len = max(len, right - left + 1);
                right++;
                if(right < size && A[right] == 1){
                    count++;
                }
            }
            else{
                if(A[left] == 1){
                    count--;
                }
                left++;
            }
        }
        return len;
    }
};